src/lib.rs

```diff
@@ -27,102 +28,1090 @@ pub extern "C" fn wapc_init() {
     register_function("protocol_version", protocol_version_guest);
 }
 
+fn is_valid_id(id: i64, ranges: &[IDRange]) -> bool {
+    for range in ranges {
+        if id >= range.min && id <= range.max {
+            return true;
+        }
+    }
+    false
+}
+
+fn enforce_run_as_user_rule(
+    security_context_option: Option<apicore::SecurityContext>,
+    validation_request: &ValidationRequest<Settings>,
+) -> Result<Option<apicore::SecurityContext>, String> {
+    let mut security_context = security_context_option.unwrap_or(apicore::SecurityContext {
+        capabilities: None,
+        allow_privilege_escalation: None,
+        privileged: None,
+        proc_mount: None,
+        read_only_root_filesystem: None,
+        run_as_group: None,
+        run_as_non_root: None,
+        run_as_user: None,
+        se_linux_options: None,
+        seccomp_profile: None,
+        windows_options: None,
+    });
+    if validation_request.settings.run_as_user.rule == "MustRunAs" {
```

Comment:
It's common to use the construction `match` in this case:

```rust
match validation_request.settings.run_as_user.rule {
  "MustRunAs" => {},
  "MustRunAsNonRoot" => {},
}
```

Reply:
To make this work I need to use the `as_str()` function. Is this okay? I mean, it sounds weird to me. But it may be my lack of XP with Rust... Let me update the PR and we can continue the discussion.